Library/Homebrew/cask/lib/hbc/installer.rb

```diff
@@ -420,6 +436,14 @@ def zap
       purge_caskroom_path
     end
 
+    def backup_path(path)
+      Pathname.new "#{path}.upgrading" unless path.nil?
```

Comment:
When would `path` be `nil`?

Reply:
Also, since this is only used for `staged_path`, you could make this take no arguments and just return `Pathname("#{@cask.staged_path}.upgrading")`.

Reply:
`path` should never be  `nil`, but I wanted to preserve the semantics of `gain_permissions_remove(staged_path) if !staged_path.nil? && staged_path.exist?`.